ospd_openvas/preferencehandler.py

```diff
@@ -379,37 +383,52 @@ def build_alive_test_opt_as_prefs(
 
     def prepare_alive_test_option_for_openvas(self):
-        """ Set alive test option. Overwrite the scan config settings.
-        Check if test_alive_hosts_only feature of openvas is active.
-        If active, put ALIVE_TEST enum in preferences. """
+        """ Set alive test option. Overwrite the scan config settings."""
+        settings = Openvas.get_settings()
+        if settings and self.target_options.get('alive_test'):
+            alive_test_opt = self.build_alive_test_opt_as_prefs(
+                self.target_options
+            )
+            self.kbdb.add_scan_preferences(
+                self._openvas_scan_id, alive_test_opt
+            )
+
+    def prepare_boreas_alive_test(self):
+        """ Set alive_test for Boreas if
+        test_alive_hosts_only scanner config was set"""
         settings = Openvas.get_settings()
+        alive_test = -1
+
         if settings:
             test_alive_hosts_only = settings.get('test_alive_hosts_only')
-            if test_alive_hosts_only:
-                if self.target_options.get('alive_test'):
-                    try:
-                        alive_test = int(self.target_options.get('alive_test'))
-                    except ValueError:
-                        logger.debug(
-                            'Alive test settings not applied. '
-                            'Invalid alive test value %s',
-                            self.target_options.get('alive_test'),
-                        )
-                    # Put ALIVE_TEST enum in db, this is then taken
-                    # by openvas to determine the method to use
-                    # for the alive test.
-                    if alive_test >= 1 and alive_test <= 31:
-                        item = 'ALIVE_TEST|||%s' % str(alive_test)
-                        self.kbdb.add_scan_preferences(
-                            self._openvas_scan_id, [item]
-                        )
-            elif self.target_options.get('alive_test'):
-                alive_test_opt = self.build_alive_test_opt_as_prefs(
-                    self.target_options
-                )
-                self.kbdb.add_scan_preferences(
-                    self._openvas_scan_id, alive_test_opt
-                )
+            if not test_alive_hosts_only:
+                return
+            alive_test_str = self.target_options.get('alive_test')
+            if alive_test_str is not None:
+                try:
+                    alive_test = int(alive_test_str)
+                except ValueError:
+                    logger.debug(
+                        'Alive test preference for Boreas not set. '
+                        'Invalid alive test value %s.',
+                        alive_test_str,
+                    )
+            # ALIVE_TEST_SCAN_CONFIG_DEFAULT if no alive_test provided
+            else:
+                alive_test = AliveTest.ALIVE_TEST_SCAN_CONFIG_DEFAULT
+
+        if 1 <= alive_test <= 31:
```

Comment:
What does this check mean? Please add a comment or even better add constants for 1 and 31 too. Btw. I am not sure if the check works. Maybe it must be `if <= alive_test and alive_test <= 31`

Reply:
`alive_test` is a bit mask. If an `alive_test` was set then the max value will be 11111 (31, all alive_test chosen) and min 1 (only first test chosen). So legal values if an `alive_test` was set are between 1 and 31.

Reply:
At the moment this line is not readable to understand its purpose. So at least it needs a comment. Better even e.g. if it is used to check the validity of the alive_test value then extract it to a function e.g.
```python
def is_valid_alive_test(self, alive_test):
    return alive_test >= ALIVE_TEST_MIN_VALUE and alive_test <= ALIVE_TEST_MAX_VALUE
```

Reply:
Sound good. I will do that in a different PR as the comparison is already made in other functions too.